lib/ecto_morph.ex

```diff
@@ -478,69 +506,88 @@ defmodule EctoMorph do
   """
   # Now the natural question is can we extend this to allow the Repo.preload syntax ? I.e. a tree?
   # [:this, plus: :this, and: [:also, these: :too]] ? I think the zipper Idea helps that along a lot.
+
+  # so let's extract the traversal. And let's have it be a keyword lens maybe. The simplest way
+  # is to expand the lens and traverse it then. But after extracting the traversal.
   def validate_nested_changeset(_, [], _) do
     raise InvalidPathError, "You must provide at least one field in the path"
   end
 
-  def validate_nested_changeset(changeset, path_to_nested_changeset, validation_fun) do
-    walk_the_path({[{nil, changeset}], path_to_nested_changeset}, validation_fun)
+  def validate_nested_changeset(changeset, path, validation) do
```

Comment:
We should say in the docs that this will not validate a non existing schema (as if the schema pointed to in the path is not a change on the changeset then the validation function wont run at all.